x/cdp/keeper/draw.go

```diff
@@ -200,10 +197,16 @@ func (k Keeper) ReturnCollateral(ctx sdk.Context, cdp types.CDP) {
 	}
 }
 
-func (k Keeper) calculatePayment(ctx sdk.Context, fees sdk.Coins, payment sdk.Coins) (sdk.Coins, sdk.Coins) {
+func (k Keeper) calculatePayment(ctx sdk.Context, owed sdk.Coins, fees sdk.Coins, payment sdk.Coins) (sdk.Coins, sdk.Coins) {
 	// divides repayment into principal and fee components, with fee payment applied first.
+
 	feePayment := sdk.NewCoins()
 	principalPayment := sdk.NewCoins()
+	overpayment := sdk.NewCoins()
+	if payment.IsAnyGT(owed) {
```

Comment:
`IsAllGT`? could be a negative coins panic in the if block

Reply:
Kind of a moot point, since we're only allowing a single stable coin at this point, but neither `IsAnyGT` nor `IsAllGT` is safe. For example, if the payment due was [500usdx, 500susd] and the user input [600usdx,400susd], `IsAny` would panic, and `IsAll` would overpay. The proper was is to compare each denom and then subtract only the ones that are overpaying. 

I'll change to `IsAll` for now and leave a TODO if/when we support multiple stable denoms.